src/conversions/num_rational.rs

```diff
@@ -0,0 +1,253 @@
+//!     m.add_function(wrap_pyfunction!(add_five_to_fraction, m)?)?;
+//!     Ok(())
+//! }
+//! ```
+//!
+//! Python code that validates the functionality:
+//! ```python
+//! from my_module import add_five_to_fraction
+//! from fractions import Fraction
+//!
+//! fraction = Fraction(2,1)
+//! fraction_plus_five = add_five_to_fraction(f)
+//! assert fraction + 5 == fraction_plus_five
+//! ```
+
+use crate::ffi;
+use crate::sync::GILOnceCell;
+use crate::types::any::PyAnyMethods;
+use crate::types::PyType;
+use crate::{Bound, FromPyObject, IntoPy, Py, PyAny, PyObject, PyResult, Python, ToPyObject};
+use std::os::raw::c_char;
+
+#[cfg(feature = "num-bigint")]
+use num_bigint::BigInt;
+use num_rational::Ratio;
+
+static FRACTION_CLS: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+
+fn get_fraction_cls(py: Python<'_>) -> PyResult<&Bound<'_, PyType>> {
+    FRACTION_CLS.get_or_try_init_type_ref(py, "fractions", "Fraction")
+}
+
+macro_rules! rational_conversion {
+    ($int: ty) => {
+        impl<'py> FromPyObject<'py> for Ratio<$int> {
+            fn extract_bound(obj: &Bound<'py, PyAny>) -> PyResult<Self> {
+                let py = obj.py();
+                let py_numerator_obj = unsafe {
+                    Bound::from_owned_ptr_or_err(
+                        py,
+                        ffi::PyObject_GetAttrString(
+                            obj.as_ptr(),
+                            "numerator\0".as_ptr() as *const c_char,
+                        ),
+                    )
+                };
+                let py_denominator_obj = unsafe {
+                    Bound::from_owned_ptr_or_err(
+                        py,
+                        ffi::PyObject_GetAttrString(
+                            obj.as_ptr(),
+                            "denominator\0".as_ptr() as *const c_char,
+                        ),
+                    )
+                };
+                let numerator_owned = unsafe {
+                    Bound::from_owned_ptr_or_err(
+                        py,
+                        ffi::PyNumber_Long(py_numerator_obj?.as_ptr()),
+                    )?
+                };
+                let denominator_owned = unsafe {
+                    Bound::from_owned_ptr_or_err(
+                        py,
+                        ffi::PyNumber_Long(py_denominator_obj?.as_ptr()),
+                    )?
+                };
+                let rs_numerator: $int = numerator_owned.extract()?;
+                let rs_denominator: $int = denominator_owned.extract()?;
+                Ok(Ratio::new(rs_numerator, rs_denominator))
+            }
+        }
+
+        impl ToPyObject for Ratio<$int> {
+            fn to_object(&self, py: Python<'_>) -> PyObject {
+                let fraction_cls = get_fraction_cls(py).expect("failed to load fractions.Fraction");
+                let ret = fraction_cls
+                    .call1((self.numer().clone(), self.denom().clone()))
```

Comment:
I think you might need to use `call` here now (except if `fractions.Fraction` can also be constructed from a tuple). This should also be caught by the tests (if not, we need to add one)

Reply:
Please correct me if I am wrong, but in this context it makes no difference I believe, since `call1` is the python expresion `self(*args)` and `call` is the same except it adds supports for `self(*args,**kwargs)`. Also, they both have `*args` which must `impl IntoPy<Py<PyTuple>>`, or at least what I see from the [docs](https://docs.rs/pyo3/0.21.2/pyo3/types/trait.PyAnyMethods.html#tymethod.call). 

I added another test to show that a tuple is still not supported for `fractions.Fraction`.